### ISSUE TITLE
Should techniques for minimum contrast recommend alternate versions?

### ISSUE THREAD

**Issue by alastc** [w3c/wcag#1176]
Posting on behalf of Gundula:
<hr>
when looking over the color contrast relevant Success Criteria and their understanding pages I came over some techniques for the minimum contrast ([understanding doc](https://www.w3.org/WAI/WCAG21/Understanding/contrast-minimum.html)) which I would like to discuss.
 
In technique [G174](https://www.w3.org/WAI/WCAG21/Techniques/general/G174) it says:

> "When the contrast between the text and its background for some portion of the page has not been designed to meet the contrast level for Success Criterion 1.4.3 or 1.4.6, it is possible to meet these guidelines using the "Alternate Version" clause in the conformance requirements (Conformance Requirement 1). […]"
 
In the sequel of the text, a high contrast theme is explicitly mentioned as the only alternative.
 
As a consequence, a provider of software with a web based User Interface can reach compliance by providing a very shady and soft colored theme not fulfilling any color contrast consideration and a high contrast theme (in most cases high contrast white on black) as alternative.

I feel this does not really meet user's needs. A high contrast theme is very harsh and painful for most eyes with good color vision. A shady theme is not well perceptible for most users in bright environments or with glare on the screen.

In addition we observed users going away from high contrast white-on-black when they find a dark theme (dark background with brighter text fulfilling minimum contrast but not high contrast).

This matches well the elaboration done and research cited on [uxmovement](https://uxmovement.com/content/why-you-should-never-use-pure-black-for-text-or-backgrounds/).
 
So what is the rationale behind that recommendation in G174?

Shouldn't it be changed to oblige authors and software providers to offer a default theme that fulfills minimum contrast (level AA) and in addition a high contrast theme fulfilling the AAA success Criterion?

Or to offer at least a minimum contrast theme plus a high contrast theme as alternatives complying to 1.4.3 and 1.4.6, respectively, if the default theme does not comply?

I suggest to change G174 to request a minimum contrast theme as alternative if 1.4.3 is not fulfilled and a high contrast theme as alternative if 1.4.6 is not fulfilled.

**Comment by detlevhfischer** [comment]
Before the German BITV-Test fully aligned with WCAG, we required at least 3:1 text contrast for the default version in cases where the author relied on style switchers to meet 4.5:1. 
A common case in the wild seems to be authors using contrast below 4.5:1 (but usually not VERY soft contrast, more in the range of 2.5:1 to 4.5:1) for bits of text (headings, background colors) to reflect the color scheme of the sote's brand or logo, and then having a style switcher to crank up contrast somewhat without aiming to provide a harsh high contrast mode. I would agree to scrapping the alternate version loophole, or at least requiring a somewhat relaxed ratio for the default version, but this may get a lot of pushback form authors (echoing site owners' wishes).

**Comment by patrickhlauke** [comment]
> In the sequel of the text, a high contrast theme is explicitly mentioned as the only alternative.

techniques are not exhaustive and only informative. nothing is stopping anybody from going above and beyond individual techniques. there are two examples for G174 https://www.w3.org/WAI/WCAG21/Techniques/general/G174#examples - the first one suggesting a switcher that goes to an alternative that meets the minimum 4.5:1

> Shouldn't it be changed to oblige authors and software providers to offer a default theme that fulfills minimum contrast (level AA) and in addition a high contrast theme fulfilling the AAA success Criterion?

techniques don't oblige anybody. they're suggestions, ideas, one of many ways of achieving something. what counts is fulfilling the requirements of the SCs, not following techniques.

> A high contrast theme is very harsh and painful for most eyes with good color vision

I don't believe that when the technique talks about high contrast (and in the second example it even puts it in quotes ... "high contrast") it refers to any particular type of theme other than a theme that has a high enough contrast to pass the criterion. i.e. what you'd refer to as a dark theme could be seen as a "high contrast" theme since it has...high contrast. there's no specific mention or implication in that technique that says it's a "white on black" type extreme theme. and even if it was, again that technique is only informative.

Long story short: as it's an informative technique, we can add even more examples. for instance, an extra example that says "the site offers the user a style switcher, with a selection of various themes, including a 'dark theme' and a completely "white on black" theme". that sort of addition would be non-controversial and easy to put through due to the informative/non-normative nature of techniques.

**Comment by gundulaniemann** [comment]
Thank you for the further input.

I understand the wording 'high contrast' is not fixed and can be understood in different ways. It may be used to refer to contrasts fulfilling 1.4.6 (which is often "white-on-black" with some few marking colors, compare the wording in the Windows OS).
Yet in this communication as well as in technique G174 it is used to refer to a theme fulfilling 1.4.3. Is that correct?

Though techniques do not oblige, they do define what (among various alternatives) is seen as sufficient to fulfill the Success Criterion.

I think it might not meet most users needs to have available one theme that does not fulfill 1.4.3, and one alternative that fulfills 1.4.6 and 1.4.3 at the same time.
He/she might lack the greater variety in colors and luminance possible in a thme fulfilling 1.4.3, yet having a contrast between 4.5:1 and 7.0:1 in some places.

Nevertheless I understand from your responses and some feedback I received per mail, that fulfilling 1.4.6 implies fulfilling 1.4.3.
Is that right?

I like the idea to suggest to use various themes, as Patrick Lauke proposed above.
I propose the following wording for a new example:
"the site offers the user a style switcher with a selection of various themes, including a 'dark theme' fulfilling 1.4.3 and a 'white on black theme' fulfilling 1.4.6".
